lisa/sut_orchestrator/libvirt/ch_platform.py

```diff
@@ -176,6 +176,22 @@ def _create_node_os_disk(
         self, environment: Environment, log: Logger, node: Node
     ) -> None:
         node_context = get_node_context(node)
+        if environment.runbook.nodes_requirement:
+            node_req: List[schema.NodeSpace] = environment.runbook.nodes_requirement
+            for _, node_space in enumerate(node_req):
+                node_runbook: BaseLibvirtNodeSchema = node_space.get_extended_runbook(
```

Comment:
I believe you can access the runbook via `self.runbook`.

Reply:
self.runbook will have platform wide runbook. For node (requirement) specific runbook, this is the only way I guess https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/libvirt/platform.py#L381. Perhaps we can add disk size in `NodeContext` to get the disk size here.

Reply:
Yeah, if it is too difficult to access the runbook, then it makes sense to add it to `NodeContext`. (Doing this also makes it consistent with the other properties.)

Reply:
Yes Chi, that is what i am using to get the custom runbook under _create_node_os_disk()

Reply:
While `NodeContext` is technically for dynamic information, please just use it anyway. This code is a mess. It is a lot simpler to just copy the relevant information into `NodeContext` during `_configure_node`.

Reply:
Used NodeContext to get resize parameter